docs/machine-learning/ml-tpm-guidance-blog.md

```diff
@@ -0,0 +1,99 @@
+
+- Focus on **System Design** principles to identify the architectural components, entities, interfaces, constraints. Ask the right questions early and explore design alternatives with the engineering team. [Mona Soliman Habib](https://www.linkedin.com/in/msolhab/), Principal Data and Applied Scientist at Microsoft considers this to be one of the core fundamentals of a ML Focused project where TPMs can have significant impact.
+
+- Think hard about the **costs of ML** and whether you can solve a repetitive problem at scale. Many a times, you can solve customer problems with data analytics, dashboards, or rule-based algorithms. I know not so cool, but true!
+
+### Taming the ambiguity beast
+
+Ok, so you figured, Machine Learning might help in your project, now what?
+
+ML projects are plagued with a phenomenon we can call as the "**Death by Unknowns**". It's like you were told that angels will meet you at the end of a dark tunnel, but no one mentioned that there are Merpeople, Trolls and a Basilisk waiting for you (Harry Potter fans are smiling :)). Unlike software engineering projects, ML focused projects can result in quick success early (aka sudden decrease in error rate), but this may flatten eventually. Few things to consider:
+
+- The most important thing as TPM is to **set clear expectations**. Identify the performance metrics and discuss on a "good enough" prediction rate that will bring value to the business. An 80% "good enough" rate may save business costs and increase productivity but if going from 80 to 95% would require unimaginable amount of cost and effort. Is it worth it? Can it be a progressive road map?
+
+> If the customer wants \> 95% accuracy with limited data and time, they are living in Lala land.
+  
+- Create a smaller team and **undertake a feasibility analysis** through techniques like [EDA](https://en.wikipedia.org/wiki/Exploratory_data_analysis) (Exploratory Data Analysis). A [feasibility study](https://microsoft.github.io/code-with-engineering-playbook/machine-learning/ml-feasibility-study/) is much cheaper to evaluate data quality, customer constraints and model feasibility. It allows a TPM to better understand customer use cases and current environment and can act as a fail-fast mechanism. Note that feasibility should be shorter (in weeks) else it misses the point of saving costs.
+
+- As in any project, there will be new needs (additional data sources, technical constraints, hiring data labelers, business users time etc.), factor these in your cost and schedule to avoid surprises.
+
+### Notebooks != ML Production
```

Comment:
Production aspects are not the only limit of notebooks. Reproducibility and collaboration between team members are also a reason to go with a more structured approach.

Reply:
agreed, incorporated this in the main md